### PR TITLE
tree-sitter-grammar.eclass: force use of local parser.h

### PR DESCRIPTION
**Pull request by thesamesam**
There shouldn't be a system-wide copy of parser.h and upstream plan on
dropping this from dev-libs/tree-sitter as it can cause issues if there's
a mismatch between the version used for the bundled generated parser
vs the header used to build the library.

Force use of the correct one in the dist tarball at src/tree_sitter/parser.h
instead. Drop the dependency on dev-libs/tree-sitter given we were only depending
on it for this header.

Bug: tree-sitter/tree-sitter-bash#199
Bug: https://bugs.gentoo.org/912716
Signed-off-by: Sam James <sam@gentoo.org>

### CODE REVIEW

**Review by MatthewGentoo — APPROVED** [review]
ACK for the tree-sitter changes. `bash-ts-mode` working without eating all of my memory now :)

Do all of the grammars need to be rev-bumped after this?

File: dev-libs/tree-sitter/tree-sitter-0.20.8-r1.ebuild
Comment (arthurzam) [inline dev-libs/tree-sitter/tree-sitter-0.20.8-r1.ebuild:1]: Please don't touch the version cap on pkgcheck - we have quite a lot of regressions, and it is scary for me currently without knowing the full impact of this beforehand (imagine huge amount of weird results and such - we break the trust in pkgcheck).

File: dev-util/pkgcheck/pkgcheck-0.10.25-r2.ebuild
Comment (arthurzam) [inline dev-util/pkgcheck/pkgcheck-0.10.25-r2.ebuild:1]: Please revert for pkgcheck, so the version cap remains. We have a big amount of regressions in parsing, and I'm concerned with breaking stuff in weird places. Imagine someone gets weird unknown result without any reason - we will break the devs trust in pkgcheck. So for now, lets not jump for pkgcheck.

**Comment by arthurzam** [comment]
> ACK for the tree-sitter changes. `bash-ts-mode` working without eating all of my memory now :)
> 
> Do all of the grammars need to be rev-bumped after this?

I don't think so. We only need to verify any tree-sitter grammer bumped since https://github.com/tree-sitter/tree-sitter/commit/4a007259fc92227df4821c07463421f00428c51a entered master, so only since last 3 weeks. Any older version won't break, and younger can just be tested if they leak.

Older grammer versions work well with tree-sitter released and master, so we are safe going forward.

**Comment by thesamesam** [comment]
Merged!

**Pull request closed without merging**